aten/src/ATen/cpu/vec256/vec256_bool.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <ATen/cpu/vec256/intrinsics.h>
+#include <ATen/cpu/vec256/vec256_base.h>
+
+namespace at {
+namespace vec256 {
+namespace {
+
+#ifdef __AVX2__
+
+template <>
+struct Vec256<uint8_t> {
+  static constexpr int size() {
+    return 32;
+  }
+};
+
+template <>
+Vec256<uint8_t> inline operator&&(
+    const Vec256<uint8_t> &a, const Vec256<uint8_t> &b) {
+  Vec256<uint8_t> c = Vec256<uint8_t>();
+
+  __m256i ma = _mm256_loadu_si256(reinterpret_cast<const __m256i*>(a));
+  __m256i mb = _mm256_loadu_si256(reinterpret_cast<const __m256i*>(b));
+
+  ma = _mm256_or_si256(ma, _mm256_srli_si256(ma, 4));
+  ma = _mm256_or_si256(ma, _mm256_srli_si256(ma, 2));
+  ma = _mm256_or_si256(ma, _mm256_srli_si256(ma, 1));
+
+  mb = _mm256_or_si256(mb, _mm256_srli_si256(mb, 4));
+  mb = _mm256_or_si256(mb, _mm256_srli_si256(mb, 2));
+  mb = _mm256_or_si256(mb, _mm256_srli_si256(mb, 1));
+
+  _mm256_storeu_si256(
```

Comment:
Don't need a mask here, as the [doc](https://software.intel.com/en-us/node/523973) says the high-order bits will be cleared to 0.  (BTW, I think there are some typos in the doc. e.g., low-order -> high-order)

Will add this to comments, in final revision.